### PR TITLE
Allow handling of vendor change option 

### PR DESCRIPTION
**Pull request by mseidl**
Handles allow vendor change when passed from the server now.

### CODE REVIEW

File: actions/packages.py [hunk 1]
```diff
@@ -163,25 +163,27 @@ def patch(self, updatestack_only=False):
...
-    def dup(self, channel_names=None, dry_run=False):
+    def dup(self, channel_names=None, dry_run=False, allow_vendor_change=False):
         args = ["-n", "-x", "dup", "--auto-agree-with-licenses"]
 
         if self.download_only:
             args.append("--download-only")
         if dry_run:
             args.append("--dry-run")
+        if allow_vendor_change and self.dup_version == 2:
+            args.append("--allow-vendor-change")
```
Comment (mcalmer) [inline actions/packages.py:174]:
```suggestion
```

File: actions/packages.py [hunk 2]
```diff
@@ -163,25 +163,27 @@ def patch(self, updatestack_only=False):
...
         if self.dup_version == 1:
             if channel_names and type(channel_names) == type([]):
                 for name in channel_names:
                     args.append("--from")
                     args.append("spacewalk:%s" % name)
-        elif self.dup_version == 2:
+        elif not allow_vendor_change and self.dup_version == 2:
```
Comment (mcalmer) [inline actions/packages.py:180]:
```suggestion
        elif self.dup_version == 2:
            if allow_vendor_change:
                args.append("--allow-vendor-change")
            else:
```

**Review by mcalmer — CHANGES_REQUESTED** [review]
What do you think about the suggested changes.

File: actions/packages.py (hunk 1 above)
Comment (mcalmer) [reply actions/packages.py:174]: Drop this part, we do it later. See below.

File: actions/packages.py [hunk 3]
```diff
@@ -163,25 +163,30 @@ def patch(self, updatestack_only=False):
...
-    def dup(self, channel_names=None, dry_run=False):
+    def dup(self, channel_names=None, dry_run=False, allow_vendor_change=False):
         args = ["-n", "-x", "dup", "--auto-agree-with-licenses"]
 
         if self.download_only:
             args.append("--download-only")
         if dry_run:
             args.append("--dry-run")
+        if allow_vendor_change and self.dup_version == 2:
+            args.append("--allow-vendor-change")
```
Comment (mcalmer) [inline actions/packages.py:174]: This can be removed

File: actions/packages.py [hunk 4]
```diff
@@ -163,25 +163,30 @@ def patch(self, updatestack_only=False):
...
         if self.dup_version == 1:
             if channel_names and type(channel_names) == type([]):
                 for name in channel_names:
                     args.append("--from")
                     args.append("spacewalk:%s" % name)
         elif self.dup_version == 2:
+            if allow_vendor_change:
+                args.append("--allow-vendor-change")
+            else:
             args.append("--no-allow-vendor-change")
```
Comment (mcalmer) [inline actions/packages.py:184]: The indentation is wrong.

**Review by mcalmer — CHANGES_REQUESTED** [review]

**Review by mcalmer — APPROVED** [review]
Looks good now

**Review by mcalmer — CHANGES_REQUESTED** [review]
Ahh, one problem found. We do not have the tooling in this project as we have in uyuni git.
Here the changelog should have the header.

**Review by mcalmer — APPROVED** [review]

**Review by meaksh — APPROVED** [review]

**Pull request merged**